integration/data/test_basic_ops.py

```diff
@@ -42,31 +44,35 @@ def test_basic_rw(dev):  # NOQA
         verify_data(dev, offset, data)
 
 
-def test_rw_with_metric(grpc_controller,  # NOQA
-                        grpc_replica1, grpc_replica2):  # NOQA
-    rw_dev = get_dev(grpc_replica1, grpc_replica2, grpc_controller)
+def test_rw_with_metric(grpc_replica1, grpc_replica2, grpc_controller):  # NOQA
+    rw_dev = get_dev(grpc_replica1, grpc_replica2,
+                     grpc_controller)
 
-    replies = grpc_controller.metric_get()
-    # skip the first metric since its fields are 0
-    next(replies).metric
-
-    for i in range(0, 5):
+    for i in range(0, 1):
         base = random.randint(1, SIZE - PAGE_SIZE)
         offset = (base // PAGE_SIZE) * PAGE_SIZE
         length = base - offset
         data = random_string(length)
         verify_data(rw_dev, offset, data)
 
+        iter = 0
         while 1:
+            iter += 1
             try:
-                metric = next(replies).metric
+                metrics = grpc_controller.metrics_get()
+                metrics = metrics.metrics
+
                 # it's hard to confirm the accurate value of metric
-                assert metric.readBandwidth != 0
-                assert metric.writeBandwidth != 0
-                assert metric.iOPS != 0
+                assert metrics.readThroughput != 0
+                assert metrics.writeThroughput != 0
+                assert metrics.readIOPS != 0
+                assert metrics.writeIOPS != 0
                 break
-            except StopIteration:
-                time.sleep(1)
+            except Exception as e:
+                if iter < METRICS_RETRY_COUNTS:
```

Comment:
Why need retry count? is it because the runtime record not updated yet?

Reply:
https://github.com/longhorn/longhorn-engine/pull/794#discussion_r1050728159